Libraries/LibWeb/CSS/SelectorEngine.cpp

```diff
@@ -759,6 +760,140 @@ static inline bool matches_pseudo_class(CSS::Selector::SimpleSelector::PseudoCla
 
         return false;
     }
+    case CSS::PseudoClass::Valid: {
```

Comment:
There's a lot of duplication between `:valid/:invalid`, and `:user-valid/:user-invalid`. I'd be tempted to try and combine those, but you don't have to.

Reply:
I feel like that'd make it more unreadable, since they all have some checks somewhere in the middle flipped and elements can be valid, invalid or neither...

I think this way it's just more straightforward and it's not like the code is, or probably will ever, be duplicated in many more places.
Plus it's duplicated in the spec as well.